pkg/run/install_bucket_source_and_ks.go

```diff
@@ -164,3 +172,66 @@ func CleanupBucketSourceAndKS(log logger.Logger, kubeClient *kube.KubeHTTP, name
 
 	return nil
 }
+
+// FindConditionMessages finds the messages in the condition of objects in the inventory.
+func FindConditionMessages(kubeClient *kube.KubeHTTP, ks *kustomizev1.Kustomization) ([]string, error) {
```

Comment:
I think it would be easier to understand this function with some tests serving as examples - could we add some?

Reply:
sure.